tests/core/middleware/test_name_to_address_middleware.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from ens import ENS
+from web3 import Web3
+from web3.exceptions import (
+    InvalidAddress,
+)
+from web3.middleware import (  # noqa: F401
+    construct_fixture_middleware,
+    name_to_address_middleware,
+)
+from web3.providers.base import (
+    BaseProvider,
+)
+
+
+@pytest.fixture
+def w3():
+    w3 = Web3(providers=[BaseProvider()], middlewares=[])
+    w3.ens = ENS.fromWeb3(w3)
+    w3.middleware_stack.add(name_to_address_middleware(w3))
+    return w3
+
+
+def test_pass_name_resolver(w3):
+    return_chain_on_mainnet = construct_fixture_middleware({
+        'net_version': '1',
+    })
+    w3.middleware_stack.inject(return_chain_on_mainnet, layer=0)
+    with pytest.raises(NotImplementedError):
+        w3.eth.getBalance('ethereum.eth')
+
+
+def test_fail_name_resolver(w3):
+    return_chain_on_mainnet = construct_fixture_middleware({
+        'net_version': '2',
+    })
+    w3.middleware_stack.inject(return_chain_on_mainnet, layer=0)
+    with pytest.raises(InvalidAddress):
```

Comment:
This test can now confirm that the name shows up in the exception with something like:

```py
with pytest.raises(InvalidAddress, match='.*ethereum\.eth.*'):
```

Reply:
Done